### ISSUE TITLE
Staging optional input files using AWS Batch

### ISSUE THREAD

**Issue by stekaz** [nextflow-io/nextflow#1233]
## Bug report

TLDR: the [optional input pattern](https://nextflow-io.github.io/patterns/index.html#_optional_input) works as expected when run locally, but will fail when attempting to stage missing input files to a remote filesystem/object store.

### Expected behavior and actual behavior

I think failing this way _is_ the expected behaviour when attempting to stage missing files to s3, but it would be great if nextflow could better handle the staging of optional input files somehow. Perhaps a syntax change is needed like what is proposed using DSL2 #1201.

The workaround I have is to use a string instead of a file object. For example:

```
params.inputs = 'prots/*{1,2,3}.fa'
params.filter = 'NO_FILE'

prots_ch = Channel.fromPath(params.inputs)
opt_file = params.filter // look mom, no hands!

process foo {
  input:
  file seq from prots_ch
  file opt from opt_file

  script:
  def filter = opt.name != 'NO_FILE' ? "--filter $opt" : ''
  """
  your_commad --input $seq $filter
  """
}
```

This has the unfortunate side-effect of creating and staging a whole lot of temporary files in the work-dir/tmp folder. But it appears to work.

### Steps to reproduce the problem

For example, create your config for AWS Batch in the usual way:

```
aws {

  region = 'my-region'
}

executor.awscli = '/home/ec2-user/miniconda/bin/aws'

process {

  container = 'ubuntu:18.04'
  executor = 'awsbatch'
  queue = 'my-queue'
}
```

Change the `region` and `queue` definitions as required. Then, run the `patterns/optional-input.nf ` project specifying your s3 work-dir:

```
$ nextflow -C awsbatch.config run patterns/optional-input.nf -w s3://testbucket/test
```

### Program output 

```
N E X T F L O W  ~  version 19.04.1
Pulling nextflow-io/patterns ...
Already-up-to-date
Launching `nextflow-io/patterns` [sad_monod] - revision: 8f11f7ee61 [master]
[warm up] executor > awsbatch
ERROR ~ Error executing process > 'foo (1)'

Caused by:
  Can't stage file file:///home/ec2-user/test/NO_FILE -- file does not exist

Command executed:

  echo your_commad --input prot_2.fa

Command exit status:
  -

Command output:
  (empty)

Work dir:
  s3://testbucket/test/66/d7d8102e230963fdfc173bf9006406

Tip: view the complete command output by changing to the process work dir and entering the command `cat .command.out`

 -- Check '.nextflow.log' file for details
[66/d7d810] process > foo [  0%] 2 of 0, failed: 2
```

### Environment 

* Nextflow version: 19.04.1
* Java version: openjdk version "1.8.0_201"
* Operating system: Amazon Linux 2

**Comment by stekaz** [comment]
Actually, I think there's a better solution: basically just keep an empty file in `$baseDir` and point to it in your scripts. For example:

```
params.inputs = 'prots/*{1,2,3}.fa'
params.filter = "${baseDir}/assets/null/NO_FILE"

prots_ch = Channel.fromPath(params.inputs)
opt_file = file(params.filter)

process foo {
  input:
  file seq from prots_ch
  file opt from opt_file

  script:
  def filter = opt.name != 'NO_FILE' ? "--filter $opt" : ''
  """
  your_commad --input $seq $filter
  """
}
```

**Comment by pditommaso** [comment]
Ok, thanks for the tip. The plain is to provide a simpler way to manage optional input files.
